US018 Create a Post Operation to Create a New User

RobertoPloneda opened the pull request.

RobertoPloneda committed:
US018 Create a Post Operation to Create a New User

ramiroAT commented on src/test/java/com/at/globalclasses/MongoDBConnection.java:
```diff
@@ -87,12 +87,16 @@ public boolean compareID(String collection, String id) {
     public String getObjectByID(String collection, String id) {
         String mObject = "";
+        JSONObject json = new JSONObject(id);
+        String idValue = json.getString("id");
         MongoCollection<Document> coll = mDataBase.getCollection(collection);
-        FindIterable<Document> findIterable = coll.find(Filters.eq("_id", new ObjectId(id)));
+        ObjectId objectId = new ObjectId(idValue);
+        FindIterable<Document> findIterable = coll.find(Filters.eq("_id", objectId));
```
You could have left the method just as it was. These lines that you added are not needed.

RobertoPloneda replied:
I put it the way it was before in the new commit

ramiroAT commented on src/test/java/com/at/globalclasses/MongoDBUtils.java:
```diff
@@ -22,6 +23,16 @@ public static String getRandomID(String env, String mDataBase, String collection
+
+    public static String compareID(String env, String mDataBase, String collection, String response) {
+        MongoDBConnection db = new MongoDBConnection(env, mDataBase);
+
```
Here you can extract the id string value before to call getObjectById. Also this method is not compating anything at all so could be renamed.

RobertoPloneda replied:
I changed the name and i extracted the id from here and i send the id value to getObjectById

ramiroAT reviewed (commented):
Did you also work in POst method?

RobertoPloneda committed:
Fixes for US018 when sending a Null and comparing values

Previously when i tried to send a Null i was sending a string with the word "Null" so i changed that and now im also comparing the Request body with the document from the database with the method compareDocuments in MongoDBUtils

RobertoPloneda committed:
Conflicts resolved with master

RobertoPloneda committed:
Merge branch 'master' into Sprint2_US018_Ploneda
